packages/@aws-cdk/aws-appmesh/lib/client-policy.ts

```diff
@@ -0,0 +1,111 @@
+import { ICertificateAuthority } from '@aws-cdk/aws-acmpca';
+import * as cdk from '@aws-cdk/core';
+import { CfnVirtualNode } from './appmesh.generated';
+
+enum CertificateType {
+  ACMPCA = 'acm',
+  FILE = 'file'
+}
+
+/**
+ * Properties of TLS validation context
+ */
+export interface ClientPolicyConfig {
+  /**
+   * Represents single validation context property
+   */
+  readonly clientPolicy: CfnVirtualNode.ClientPolicyProperty;
+}
+
+/**
+ * TLS Connections with downstream server will always be enforced if True
+ */
+export interface ClientPolicyOptions {
+  /**
+   * TLS enforced if True.
+   *
+   * @default true
+   */
+  readonly enforceTls?: boolean;
```

Comment:
I think I need to update the comments in code. If the `enforceTls` is `true` the virtual node communicates with all backends using Transport Layer Security (TLS). If we only want to require the use of TLS for one or more specific ports, then we specify them in `ports`. If we don't specify any ports, TLS is enforced for all ports that the `virtualnode` would use. And the default is `True`, if a value isn't specified.[Reference](https://docs.aws.amazon.com/app-mesh/latest/APIReference/API_ClientPolicyTls.html)
So we would need `enforceTls` in case where we do not want to use TLS.

Reply:
So, to sum up:

1. `enforceTls: true` (or `undefined`, as that's the default), and `ports`: `undefined`:

TLS will be enforced for all ports.

2. `enforceTls: true` (or `undefined`, as that's the default), and `ports: [1234]`:

TLS will be enforced only for port 1234.

3. `enforceTls: false`, and `ports: undefined` (default):

TLS will not be enforced for any ports.

4. `enforceTls: false`, and `ports: [1234]`:

This seems like it's wrong, and should result in a validation error...?